addons/account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -69,33 +53,67 @@ def import_file(self, cr, uid, ids, context=None):
             'type': 'ir.actions.client',
         }
 
-    def _parse_file(self, cr, uid, data_file, context=None):
-        """ Each module adding a file support must extends this method. It processes the file if it can, returns super otherwise, resulting in a chain of responsability.
-            This method parses the given file and returns the data required by the bank statement import process, as specified below.
+    @api.model
+    def _import_file(self, data_file):
+        """ Create bank statement(s) from file
+        """
+        # The appropriate implementation module returns the required data
+        currency_code, account_number, stmts_vals = self._parse_file(data_file)
+        # Check raw data
+        self._check_parsed_data(stmts_vals)
+        # Try to find the bank account and currency in odoo
+        currency_id, bank_account_id = self._find_additional_data(
+            currency_code, account_number)
+        # Find or create the bank journal
+        journal_id = self._get_journal(
+            currency_id, bank_account_id, account_number)
+        # Create the bank account if not already existing
+        if not bank_account_id and account_number:
```

Comment:
Hi Quentin, I don't understand the full detail of the code, but I just want to mention one thing that worries me and some others.

I understand the the use case to auto-create company bank accounts/journals when importing the bank statement the first time for small companies: this facilitates configuration.

However in a setup where people manage several companies in Odoo, they can easily mix bank statement files between companies, and this can lead to creating the same bank account for different companies. So this is very dangerous and that's why Alexis disabled that behaviour in the 8.0 backport.

What do you think? Do you see a way to keep the autoconfiguration goal while keeping it safe when you have many files for different companies?